Source/relay/TourGuide/Items of the Month/2023/Cincho de Mayo.ash

```diff
@@ -0,0 +1,63 @@
+RegisterResourceGenerationFunction("IOTMCinchoDeMayoGenerateResource");
+void IOTMCinchoDeMayoGenerateResource(ChecklistEntry [int] resource_entries)
+{
+    if (lookupItem("Cincho de Mayo").available_amount() == 0) return;
+    
+    // _cinchUsed is a weird preference that actually means distance from 100% you are at in your current cinch.
+    int freeRests = __misc_state_int["free rests remaining"];
+    int cinchoRests = get_property_int('_cinchRests');
+    int cinchUsed = get_property_int('_cinchUsed');
+    
+    // Since the pref is weird, this tells you your current total cinch
+    int currentCinch = 100 - cinchUsed;
+
+    // Calculating total available cinch requires storing the degradation of rest value
+    int [int] cinchLevels = listMake(30,30,30,30,30,25,20,15,10,5);
+
+    // Reiterating current state so the while loop can update it
+    int totalCinch = 100 - cinchUsed;
+    int rest = cinchoRests;
+
+    // This while loop expands your possible cinch starting at rests you haven't used.
+    while (rest < freeRests+1)
+			{
```

Comment:
Some more tabs to clean up here, this whole block should be dedented to align with the while